diesel/src/query_builder/select_by_statement/dsl_impls.rs

```diff
@@ -0,0 +1,365 @@
+    STMT: DistinctDsl,
+    SelectByStatement<S, STMT::Output>: SelectQuery<SqlType = ST>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn distinct(self) -> Self::Output {
+        SelectByStatement::new(self.inner.distinct())
+    }
+}
+
+impl<S, STMT, Predicate> FilterDsl<Predicate> for SelectByStatement<S, STMT>
+where
+    Predicate: Expression<SqlType = Bool> + NonAggregate,
+    STMT: FilterDsl<Predicate>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn filter(self, predicate: Predicate) -> Self::Output {
+        SelectByStatement::new(self.inner.filter(predicate))
+    }
+}
+
+impl<S, STMT, Predicate> OrFilterDsl<Predicate> for SelectByStatement<S, STMT>
+where
+    Predicate: Expression<SqlType = Bool> + NonAggregate,
+    STMT: OrFilterDsl<Predicate>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn or_filter(self, predicate: Predicate) -> Self::Output {
+        SelectByStatement::new(self.inner.or_filter(predicate))
+    }
+}
+
+use crate::query_source::Table;
+
+impl<S, STMT, PK> FindDsl<PK> for SelectByStatement<S, STMT>
+where
+    STMT: FindDsl<PK>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn find(self, id: PK) -> Self::Output {
+        SelectByStatement::new(self.inner.find(id))
+    }
+}
+
+impl<ST, S, STMT, Expr> OrderDsl<Expr> for SelectByStatement<S, STMT>
+where
+    Expr: Expression,
+    Self: SelectQuery<SqlType = ST>,
+    STMT: OrderDsl<Expr>,
+    SelectByStatement<S, STMT::Output>: SelectQuery<SqlType = ST>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn order(self, expr: Expr) -> Self::Output {
+        SelectByStatement::new(self.inner.order(expr))
+    }
+}
+
+impl<ST, S, STMT, Expr> ThenOrderDsl<Expr> for SelectByStatement<S, STMT>
+where
+    Expr: Expression,
+    Self: SelectQuery<SqlType = ST>,
+    STMT: ThenOrderDsl<Expr>,
+    SelectByStatement<S, STMT::Output>: SelectQuery<SqlType = ST>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn then_order_by(self, expr: Expr) -> Self::Output {
+        SelectByStatement::new(self.inner.then_order_by(expr))
+    }
+}
+
+impl<ST, S, STMT> LimitDsl for SelectByStatement<S, STMT>
+where
+    Self: SelectQuery<SqlType = ST>,
+    STMT: LimitDsl,
+    SelectByStatement<S, STMT::Output>: SelectQuery<SqlType = ST>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn limit(self, limit: i64) -> Self::Output {
+        SelectByStatement::new(self.inner.limit(limit))
+    }
+}
+
+impl<ST, S, STMT> OffsetDsl for SelectByStatement<S, STMT>
+where
+    Self: SelectQuery<SqlType = ST>,
+    STMT: OffsetDsl,
+    SelectByStatement<S, STMT::Output>: SelectQuery<SqlType = ST>,
+{
+    type Output = SelectByStatement<S, STMT::Output>;
+
+    fn offset(self, offset: i64) -> Self::Output {
+        SelectByStatement::new(self.inner.offset(offset))
+    }
+}
+
+impl<S, STMT, Expr> GroupByDsl<Expr> for SelectByStatement<S, STMT>
```

Comment:
Reminder for myself: I need to think a bit more about the interaction of this feature with the newly merged foundation for true group by support.